archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
-    for record in resource.tree.walk:
+    args = parser.parse_args()
+    resource_id = client.get(f'/repositories/2/resources/{(args.resource_id)}').json()
+    for record in walk_tree(resource_id, client):
         updated = False
-        aojson = record.json()
-        if args.level in [record.level, "all"]:
-            notes = aojson.get("notes")
+        if args.level in [record['level'], "all"]:
+            notes = record['notes']
             for idx, note in reversed(list(enumerate(notes))):
-                if note["type"] == args.note_type:
-                    for subnote in note.get("subnotes"):
-                        content = subnote["content"]
+                if note['type'] == args.note_type:
+                    for subnote in note.get('subnotes'):
+                        content = subnote['content']
                         if contains_match(content, args.search_string):
-                            if args.action == "delete":
-                                del notes[idx]
-                                print("{} note deleted from object {}".format(
-                                    args.note_type, record.uri))
-                            if args.action == "modify":
-                                print("{} note was originally {} and was changed to {} in object {}".format(
-                                    args.note_type, content, args.replace_string, record.uri))
-                                subnote["content"] = args.replace_string
-                            log_to_spreadsheet(record)
-                            updated = True
-            if updated:
-                save_record(record.uri, aojson)
+                            handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated)
+                            
+def handle_matching_notes(args, notes, idx, record, content, subnote, client, resource_id, writer, updated):
+    if args.action == "delete":
+        del notes[idx]
+        print("{} note {} deleted from object {}".format(
+            args.note_type, args.search_string, record['uri']))
+    if args.action == "modify":
+        print("{} note was originally {} and was changed to {} in object {}".format(
+            args.note_type, content, args.replace_string, record['uri']))
+        subnote['content'] = args.replace_string
+    log_to_spreadsheet(record, writer, resource_id, client)
+    updated = True
+    if updated:
+        save_record(client, record['uri'], record)
 
 def contains_match(content, search_string):
     """Returns True if user-provided note input matches the corresponding note within a given ratio (CONFIDENCE_RATIO)."""
     ratio = fuzz.token_sort_ratio(content.lower(), search_string.lower())
     return True if ratio > CONFIDENCE_RATIO else False
```

Comment:
I think we can use the ASnake method `text_in_note` here, right? https://github.com/archivesspace-labs/ArchivesSnake/blob/master/asnake/utils/__init__.py#L106 And if I'm reading it right it might help us eliminate some of the above conditionals as well?